rules/prefer-replace-all.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+function checkReplace(node) {
+	return node.callee.property.name === 'replace';
+}
+
+function checkGlobalFlag(node) {
+	const searchPattern = node.arguments[0];
+	return Object.prototype.hasOwnProperty.call(searchPattern, 'regex') && searchPattern.regex.flags === 'g';
```

Comment:
why not just `searchPattern && searchPattern.regex && searchPattern.regex.flags === 'g'`. 
I understand flags with `i` should not pass, but how about other [flags](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#Parameters) like `gu`?

Reply:
Does the built in replace method have full unicode support? From what I'm reading it seems like it doesn't. And all of the other flags would change the search behavior as well.